From 9bd8cd6fa1fb5a1ee1324948ca1be9caab1c893d Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Sat, 13 Dec 2014 13:27:53 +0100
Subject: [PATCH 3/3] daemon: Use new functions for restoring adapters state on
 suspend.

Instead of storing state in QHash, use restoreAdaptersState() and
saveAdaptersState() introduced in previous commit.
---
 src/daemon/kded/BlueDevilDaemon.cpp | 18 ++----------------
 1 file changed, 2 insertions(+), 16 deletions(-)

diff --git a/src/daemon/kded/BlueDevilDaemon.cpp b/src/daemon/kded/BlueDevilDaemon.cpp
index 4d5fc08..7b1e6e4 100644
--- a/src/daemon/kded/BlueDevilDaemon.cpp
+++ b/src/daemon/kded/BlueDevilDaemon.cpp
@@ -66,7 +66,6 @@ struct BlueDevilDaemon::Private
     QList <DeviceInfo>                m_discovered;
     QTimer                           m_timer;
     KComponentData                  m_componentData;
-    QHash<QString, bool>            m_adapterPoweredHash;
     KSharedConfig::Ptr              m_config;
 };
 
@@ -156,23 +155,10 @@ void BlueDevilDaemon::login1PrepareForSleep(bool active)
 {
     if (active) {
         kDebug(dblue()) << "About to suspend";
-        d->m_adapterPoweredHash.clear();
-        Q_FOREACH (Adapter *adapter, Manager::self()->adapters()) {
-            kDebug(dblue()) << "Saving" << adapter->address() << adapter->isPowered();
-            d->m_adapterPoweredHash.insert(adapter->address(), adapter->isPowered());
-        }
+        saveAdaptersState();
     } else {
         kDebug(dblue()) << "About to resume";
-        QHashIterator<QString, bool> it(d->m_adapterPoweredHash);
-        while (it.hasNext()) {
-            it.next();
-            Adapter *adapter = adapterForAddress(it.key());
-            if (adapter) {
-                kDebug(dblue()) << "Restoring" << adapter->address() << it.value();
-                adapter->setPowered(it.value());
-            }
-        }
-        d->m_adapterPoweredHash.clear();
+        restoreAdaptersState();
     }
 }
 
-- 
2.1.0

